engines/wage/entities.cpp

```diff
@@ -0,0 +1,500 @@
+
+	_winningWeapons = data->readByte();
+	_winningMagic = data->readByte();
+	_winningRun = data->readByte();
+	_winningOffer = data->readByte();
+	_losingWeapons = data->readByte();
+	_losingMagic = data->readByte();
+	_losingRun = data->readByte();
+	_losingOffer = data->readByte();
+
+	_gender = data->readSByte();
+	if (data->readSByte() == 1)
+		_nameProperNoun = true;
+	else
+		_nameProperNoun = false;
+
+	_initialScene = readPascalString(data);
+	_nativeWeapon1 = readPascalString(data);
+	_operativeVerb1 = readPascalString(data);
+	_nativeWeapon2 = readPascalString(data);
+	_operativeVerb2 = readPascalString(data);
+
+	_initialComment = readPascalString(data);
+	_scoresHitComment = readPascalString(data);
+	_receivesHitComment = readPascalString(data);
+	_makesOfferComment = readPascalString(data);
+	_rejectsOfferComment = readPascalString(data);
+	_acceptsOfferComment = readPascalString(data);
+	_dyingWords = readPascalString(data);
+
+	_initialSound = readPascalString(data);
+	_scoresHitSound = readPascalString(data);
+	_receivesHitSound = readPascalString(data);
+	_dyingSound = readPascalString(data);
+
+	_weaponSound1 = readPascalString(data);
+	_weaponSound2 = readPascalString(data);
+
+	for (int i = 0; i < NUMBER_OF_ARMOR_TYPES; i++)
+		_armor[i] = NULL;
+
+	_weapon1 = NULL;
+	_weapon2 = NULL;
+
+	// Create native weapons
+	if (_nativeWeapon1.size() && _operativeVerb1.size()) {
+		_weapon1 = new Obj;
+
+		_weapon1->_name = _nativeWeapon1;
+		_weapon1->_operativeVerb = _operativeVerb1;
+		_weapon1->_type = Obj::REGULAR_WEAPON;
+		_weapon1->_accuracy = 0;
+		_weapon1->_damage = _weaponDamage1;
+		_weapon1->_sound = _weaponSound1;
+	}
+
+	if (_nativeWeapon2.size() && _operativeVerb2.size()) {
+		_weapon2 = new Obj;
+
+		_weapon2->_name = _nativeWeapon2;
+		_weapon2->_operativeVerb = _operativeVerb2;
+		_weapon2->_type = Obj::REGULAR_WEAPON;
+		_weapon2->_accuracy = 0;
+		_weapon2->_damage = _weaponDamage2;
+		_weapon2->_sound = _weaponSound2;
+	}
+
+	delete data;
+}
+
+void Chr::resetState() {
+	_context._statVariables[PHYS_STR_BAS] = _context._statVariables[PHYS_STR_CUR] = _physicalStrength;
+	_context._statVariables[PHYS_HIT_BAS] = _context._statVariables[PHYS_HIT_CUR] = _physicalHp;
+	_context._statVariables[PHYS_ARM_BAS] = _context._statVariables[PHYS_ARM_CUR] = _naturalArmor;
+	_context._statVariables[PHYS_ACC_BAS] = _context._statVariables[PHYS_ACC_CUR] = _physicalAccuracy;
+
+	_context._statVariables[SPIR_STR_BAS] = _context._statVariables[SPIR_STR_CUR] = _spiritualStength;
+	_context._statVariables[SPIR_HIT_BAS] = _context._statVariables[SPIR_HIT_CUR] = _spiritialHp;
+	_context._statVariables[SPIR_ARM_BAS] = _context._statVariables[SPIR_ARM_CUR] = _naturalArmor;
+	_context._statVariables[SPIR_ACC_BAS] = _context._statVariables[SPIR_ACC_CUR] = _physicalAccuracy;
+
+	_context._statVariables[PHYS_SPE_BAS] = _context._statVariables[PHYS_SPE_CUR] = _runningSpeed;
+}
+
+ObjArray *Chr::getWeapons(bool includeMagic) {
```

Comment:
We can simply return an `ObjArray` instead of a pointer here. This will simplify caller code by a lot.
RVO will likely be applied and avoid an unnecessary copy. And even if not, the copy should be reasonably small to not cause any issues (and if it does, we can worry about it later).